code/datums/components/basic_mob_attack_telegraph.dm

```diff
@@ -0,0 +1,82 @@
+/**
+ * Delays outgoing attacks which are directed at mobs to give players time to get out of the way
+ */
+/datum/component/basic_mob_attack_telegraph
+	/// Time to wait before attack can complete
+	var/telegraph_duration
+	/// Overlay which we display over targets
+	var/mutable_appearance/target_overlay
+	/// Our current target, if we have one
+	var/mob/living/current_target
+	/// Callback executed when we start aiming at something
+	var/datum/callback/on_began_forecast
+
+/datum/component/basic_mob_attack_telegraph/Initialize(
+	telegraph_icon = 'icons/mob/telegraphing/telegraph.dmi',
+	telegraph_state = ATTACK_EFFECT_BITE,
+	telegraph_duration = 0.3 SECONDS,
```

Comment:
0.3 is exceptionally short but really just on the verge of being dodgeable, as seen in the video, considering these are kind of a common enemy it might be a good idea to put this on 0.35 or 0.4 seconds, i dunno test and see what seems better

Reply:
My experience with testing this is that 0.3 is about right for the existing crusher gameplay (frankly, "using melee anywhere" gameplay) of ducking in, clicking, and immediately back out of range. Quite possibly other mobs would be more lenient but to me it feels right for this one. Can always be adjusted again if it turns out that I made them into killing machines, but I don't think I have and I would consider myself only of average skill at best 😅